File: Semestrul 4/MPP2/Problema8_FX_Csharp/Problema8_FX_Csharp/Utils/AgeConverter.cs
namespace Problema8_FX_Csharp.Utils;

public class AgeConverter
{
    public static int getAgeFromCNP(String CNP)
    {
        int year = int.Parse(CNP.Substring(1, 2));
        int month = int.Parse(CNP.Substring(3, 2));
        int day = int.Parse(CNP.Substring(5, 2));
        
        int currentYear = DateTime.Now.Year % 100;
        int currentMonth = DateTime.Now.Month;
        int currentDay = DateTime.Now.Day;
        
        int age = currentYear - year;
        if (month > currentMonth || (month == currentMonth && day > currentDay))
        {
            age--;
        }
        
        return age;
    }
}

File: Semestrul 4/MPP2/Problema8_SC/Problema8_SC/Problema8Model/src/main/java/problema8/model/AgeConverter.java
package problema8.model;

public class AgeConverter {
    public static int getAgeFromCNP(String cnp) {
        int year = Integer.parseInt(cnp.substring(1, 3));
        int month = Integer.parseInt(cnp.substring(3, 5));
        int day = Integer.parseInt(cnp.substring(5, 7));

        int currentYear = 2021;
        int currentMonth = 5;
        int currentDay = 10;

        int age = currentYear - year;
        if (month > currentMonth || (month == currentMonth && day > currentDay)) {
            age--;
        }

        return age;
    }
}
